release-content/0.15/migration-guides/15591_Text_rework.md

```diff
@@ -1,8 +1,28 @@
-TODO: very breaking
+The `Text` API in Bevy has been overhauled in several ways as part of Bevy 0.15.
+There are several major changes to consider:
 
-__Accessing text spans by index__
+- `ab_glyph` has been replaced with `cosmic-text`. These changes are mostly internal and the majority of users will not interact with either text backend directly.
+- each text section is now stored as a distinct entity within the standard hierarchy, rather than as a `Vec<TextSection>` on the `Text` component. Children of `Text`/`Text2d` entities with `Text` `Text2d` components will act as additional text sections.
+- like other aspects of Bevy's API, required components have replaced bundles
 
-Text sections are now text sections on different entities in a hierarchy, Use the new `TextReader` and `TextWriter` system parameters to access spans by index.
+## `TextBundle` and text styling
+
+`TextBundle` has been removed.
+Add the `Text` component to set the string displayed.
+
+`TextLayout`, `TextFont` and `TextColor` are required components for `Text`,
+and are automatically added whenever `Text` is.
+Set those values to change the text section's style.
+
+Like elsewhere in Bevy, there is no style inheritance. Consider writing your own abstraction for this if this is something you'd like to use.
```

Comment:
I can provide examples of this if interested.

Reply:
A link here would be good.

Reply:
https://github.com/viridia/thorium_ui/blob/main/crates/thorium_ui_controls/src/text_styles.rs

Reply:
```suggestion
Like elsewhere in Bevy, there is no style inheritance. Consider [writing your own abstraction for this](https://github.com/viridia/thorium_ui/blob/main/crates/thorium_ui_controls/src/text_styles.rs) if this is something you'd like to use.
```